### PR TITLE
obstacles and corners

### PR DESCRIPTION
**Pull request by CedricGuillemet**
followup on https://forum.babylonjs.com/t/navigation-mesh-and-crowd-agents/4609/86
Doc in progress

### CODE REVIEW

File: src/Navigation/INavigationEngine.ts [hunk 1]
```diff
@@ -156,12 +156,42 @@ export interface INavigationEnginePlugin {
...
     getMaximumSubStepCount(): number;
 
+    /**
+     * Creates a cylinder obstacle and add it to the navigation
+     * @param position world position
+     * @param radius cylinder radius
+     * @param height cylinder height
+     * @returns the obstacle freshly created
+     */
+    addCylinderObstacle(position: Vector3, radius: number, height: number): IObstacle;
```
Comment (Popov72) [inline src/Navigation/INavigationEngine.ts:166]: Does this signature mean the cylinder is always parallel to the Y axis? We can't have an arbitrary rotated cylinder?

File: src/Navigation/Plugins/recastJSPlugin.ts [hunk 2]
```diff
@@ -692,6 +726,25 @@ export class RecastJSCrowd implements ICrowd {
...
     }
 
+    /**
+     * Get the next corner points composing the path (max 4 points)
+     * @param index agent index returned by addAgent
+     * @returns array containing world position composing the path
+     */
+    getCorners(index: number): Vector3[]
+    {
+        var pt: number;
```
Comment (Popov72) [inline src/Navigation/Plugins/recastJSPlugin.ts:736]: `let`?

File: src/Navigation/Plugins/recastJSPlugin.ts [hunk 3]
```diff
@@ -692,6 +726,25 @@ export class RecastJSCrowd implements ICrowd {
...
+        let navPath = this.recastCrowd.getPath(index);
+        let pointCount = navPath.getPointCount();
+        var positions = [];
```
Comment (Popov72) [inline src/Navigation/Plugins/recastJSPlugin.ts:739]: `const`?

**Review by Popov72 — COMMENTED** [review]
LGTM!

File: src/Navigation/INavigationEngine.ts (hunk 1 above)
Comment (CedricGuillemet) [reply src/Navigation/INavigationEngine.ts:166]: No ellipse. But, a PI/2 on X or Z axis gives a rectangle, so you can use a box for that case ;)

**Pull request merged**